Exercise_2.java

```diff
@@ -1,40 +1,63 @@
-public class StackAsLinkedList { 
-  
+
+class StackAsLinkedList { 
+	  
     StackNode root; 
   
     static class StackNode { 
         int data; 
         StackNode next; 
-  
-        StackNode(int data) 
-        { 
-            //Constructor here 
-        } 
     } 
     
+    StackAsLinkedList(){
+    	this.root=null;
+    }
 	
     public boolean isEmpty() 
     { 
-        //Write your code here for the condition if stack is empty. 
+        return root==null;//Write your code here for the condition if stack is empty. 
+    
     } 
   
     public void push(int data) 
     { 
+    	StackNode temp=new StackNode();
+    	if(temp==null) {
+    		System.out.print("\n stack overflow");

```

Comment:
If temp is null, stack won't overflow. Stack can overflow if you are implementing it using arrays. 
@super30admin am I right?

Reply:
yup you are right